base/math.jl

```diff
@@ -911,13 +911,21 @@ function modf(x::Float64)
 end
 
 @inline function ^(x::Float64, y::Float64)
+    if isfinite(x)
+        yint = round(Int, y)
```

Comment:
I think this will throw now : `10.0^(10.0^308)`.

Reply:
Good catch. Fixed.